src/game-data/visitors/summerVisitorReducers.tsx

```diff
@@ -607,6 +611,63 @@ export const summerVisitorReducers: Record<
                 return state;
         }
     },
+    horticulturist: (state, action, pendingAction) => {
+        const horticulturistAction = pendingAction as PlayVisitorPendingAction & {
+            isDiscarding?: boolean;
```

Comment:
technically you could use the _number of cards_ chosen in the `vine` case to infer whether a discard or plant is happening, but 🤷